src/MIDebugEngine/Engine.Impl/DebuggedThread.cs

```diff
@@ -135,18 +146,60 @@ internal void MarkDirty()
             }
         }
 
-        internal void ThreadEvent(int id, bool deleted)
+        internal void ThreadCreatedEvent(int id, string groupId)
+        {
+            lock (_threadList)
+            {
+                var thread = _threadList.Find(t => t.Id == id);
+                if (thread == null)
+                {
+                    _stateChange = true;
+                }
+                if (!_threadGroups.ContainsKey(groupId))
+                {
+                    _threadGroups[groupId] = new List<int>();
+                }
+                _threadGroups[groupId].Add(id);
+            }
+        }
+
+        internal void ThreadExitedEvent(int id)
         {
             lock (_threadList)
             {
                 var thread = _threadList.Find(t => t.Id == id);
-                if ((thread != null) == deleted)
+                if (thread != null)
                 {
                     _stateChange = true;
                 }
+                foreach (var g in _threadGroups)
+                {
+                    if (g.Value.Contains(id))
+                    {
+                        g.Value.Remove(id);
+                        break;
+                    }
+                }
             }
         }
 
+        internal void ThreadGroupExitedEvent(string groupId)
+        {
+            lock (_threadList)
+            {
+                if (_threadGroups.ContainsKey(groupId))
+                {
+                    _threadGroups.Remove(groupId);
+                }
+            }
+        }
+
+        private bool IsInParent(int tid)
```

Comment:
This is going to give the wrong answer for clrdbg - we don't use thread groups. Not sure about other scenarios.